npm/pkg/transport/transport.go

```diff
@@ -110,8 +116,13 @@ func (m *Manager) handle() {
 	reflection.Register(server)
 
 	klog.Info("Starting transport manager server")
+
 	// Start gRPC Server in background
-	if err := server.Serve(lis); err != nil {
-		m.errCh <- fmt.Errorf("failed to start gRPC server: %w", err)
-	}
+	go func(errCh chan<- error) {
+		if err := server.Serve(lis); err != nil {
+			errCh <- fmt.Errorf("failed to start gRPC server: %w", err)
+		}
+	}(m.errCh)
```

Comment:
I cant recall if closures would prevent the member channel from being passed as-is into the go routine's scope or not. So I explicitly pass it in as a safety measure 🤷

Reply:
as long as it's not reassigned in a loop, the variable gets captured in the closure

Reply:
[this](https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables) is probably what you're thinking of?

Reply:
Ye I knew about that one but wasn't sure if channels might have a similar fate. Seems unlikely